src/features/swap/index.tsx

```diff
@@ -69,19 +76,63 @@ const SwapWidget = ({ sell }: Params) => {
   const dispatch = useAppDispatch()
   const isSwapFeatureEnabled = useHasFeature(FEATURES.NATIVE_SWAPS)
   const swapParams = useAppSelector(selectSwapParams)
-  const { tradeType } = swapParams
   const { safeAddress, safeLoading } = useSafeInfo()
   const [blockedAddress, setBlockedAddress] = useState('')
   const wallet = useWallet()
   const { isConsentAccepted, onAccept } = useSwapConsent()
-  // useRefs as they don't trigger re-renders
-  const tradeTypeRef = useRef<TradeType>(tradeType === 'twap' ? TradeType.ADVANCED : TradeType.SWAP)
-  const sellTokenRef = useRef<Params['sell']>(
-    sell || {
+
+  const [params, setParams] = useState<CowSwapWidgetParams>({
+    appCode: 'Safe Wallet Swaps', // Name of your app (max 50 characters)
+    width: '100%', // Width in pixels (or 100% to use all available space)
+    height: '860px',
+    chainId,
+    standaloneMode: false,
+    disableToastMessages: true,
+    disablePostedOrderConfirmationModal: true,
+    hideLogo: true,
+    hideNetworkSelector: true,
+    sounds: {
+      orderError: null,
+      orderExecuted: null,
+      postOrder: null,
+    },
+    tradeType: swapParams.tradeType === 'twap' ? TradeType.ADVANCED : TradeType.SWAP,
+    sell: sell || {
       asset: '',
       amount: '0',
     },
-  )
+    buy: {
+      asset: '',
+      amount: '0',
+    },
+    images: {
+      emptyOrders: darkMode
+        ? BASE_URL + '/images/common/swap-empty-dark.svg'
+        : BASE_URL + '/images/common/swap-empty-light.svg',
+    },
+    enabledTradeTypes: [TradeType.SWAP, TradeType.LIMIT, TradeType.ADVANCED],
+    theme: {
+      baseTheme: darkMode ? 'dark' : 'light',
+      primary: palette.primary.main,
+      background: palette.background.main,
+      paper: palette.background.paper,
+      text: palette.text.primary,
+      danger: palette.error.dark,
+      info: palette.info.main,
+      success: palette.success.main,
+      warning: palette.warning.main,
+      alert: palette.warning.main,
+    },
+    partnerFee: {
+      bps: 35,
+      recipient: SWAP_FEE_RECIPIENT,
+    },
+    content: {
+      feeLabel: 'Widget Fee',
+      feeTooltipMarkdown:
+        'The [tiered widget fee](https://help.safe.global/en/articles/178530-how-does-the-widget-fee-work-for-native-swaps) incurred here and charged by CoW DAO for the operation of the CoW Swap Widget is automatically calculated into this quote. It will contribute to a license fee that supports the Safe Community. Neither the Safe Ecosystem Foundation nor Safe (Wallet) operate the CoW Swap Widget and/or CoW Swap.',
```

Comment:
This doesn't seem to work (including formatted links into the tooltip)
<img width="361" alt="Screenshot 2024-07-01 at 11 21 48" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/9babcd5e-5684-4a12-be38-0f8cce5349c2">

Reply:
Yes, I've reported this to CoW on friday